dbms/src/Common/MPMCQueue.h

```diff
@@ -289,26 +302,33 @@ class MPMCQueue
                 return read_pos < write_pos || !isNormal();
             };
             if (!wait(lock, reader_head, node, pred, deadline))
-                return Result::TIMEOUT;
+                is_timeout = true;
         }
         if (!isCancelled() && read_pos < write_pos)
         {
             auto & obj = getObj(read_pos);
             res = std::move(obj);
             destruct(obj);
+            current_auxiliary_memory_usage -= get_auxiliary_memory_usage(res);
 
             /// update pos only after all operations that may throw an exception.
             ++read_pos;
+            if (read_pos == write_pos)
+                /// if there is no element, the current_auxiliary_memory_usage must be zero
+                current_auxiliary_memory_usage = 0;
```

Comment:
```suggestion
       assert(read_pos != write_pos || current_auxiliary_memory_usage == 0);
```

Reply:
I think it's ok to set `current_auxiliary_memory_usage =0` here, because even if `current_auxiliary_memory_usage` is not 0 due to some bugs, it still works fine after we reset it to zero.

Reply:
What about both assert and set 0?
This will catch `current_auxiliary_memory_usage`'s bug in the daily run

Reply:
OK, updated.